### PR TITLE
Add resource group id to queries table

### LINKED ISSUE #10901: Add resource group to runtime.queries system table
This is related to issue #4310

### PR DESCRIPTION
**Pull request by elonazoulay**
This is related to issue #4310
Resolves #10901

### CODE REVIEW

**Comment by raghavsethi** [comment]
I'm pretty sure this breaks the UI - could you check?

**Comment by elonazoulay** [comment]
I checked on laptop, ran queries, checked the live plan, json, stage performance and splits. Also checked the tasks and worker links from the query page and did not see any issues.

Let me know if there are other parts of the ui I can check, I'm still searching as well.

**Comment by raghavsethi** [comment]
Ah no I just meant the field that displays the resource group name on the query details page.

**Comment by elonazoulay** [comment]
Found something in webapp/assets.query.js - thanks for that!

File: presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java [hunk 1]
```diff
@@ -307,20 +307,20 @@ public void testQueryTypeBasedSelection()
...
     }
 
     @Test
     public void testClientTagsBasedSelection()
             throws InterruptedException
     {
-        assertResourceGroupWithClientTags(ImmutableSet.of("tag1"), "global.bi-user");
-        assertResourceGroupWithClientTags(ImmutableSet.of("tag1", "tag2"), "global.user-user.adhoc-user");
+        assertResourceGroupWithClientTags(ImmutableSet.of("tag1"), new ResourceGroupId(ImmutableList.of("global", "bi-user")));
+        assertResourceGroupWithClientTags(ImmutableSet.of("tag1", "tag2"), new ResourceGroupId(ImmutableList.of("global", "user-user", "adhoc-user")));
```
Comment (raghavsethi) [inline presto-tests/src/test/java/com/facebook/presto/execution/resourceGroups/db/TestQueuesDb.java]: If there are tests where this is common you can create a helper (minor suggestion)

File: presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java [hunk 2]
```diff
@@ -126,7 +126,7 @@ public void queryCreatedEvent(QueryInfo queryInfo)
                                 queryInfo.getSession().getSource(),
                                 queryInfo.getSession().getCatalog(),
                                 queryInfo.getSession().getSchema(),
-                                queryInfo.getResourceGroupName(),
+                                queryInfo.getResourceGroupId(),
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/event/query/QueryMonitor.java]: This commit title would be sufficient (no need for full message):
Replace resourceGroupName with resourceGroupId in QueryInfo
Also, squash the last commit into this one - we prefer to not have commits where Presto is broken.

File: presto-main/src/main/java/com/facebook/presto/connector/system/QuerySystemTable.java [hunk 3]
```diff
@@ -108,6 +118,17 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         return table.build().cursor();
     }
 
+    private static Block toBlock(ResourceGroupId resourceGroupId)
```
Comment (raghavsethi) [inline presto-main/src/main/java/com/facebook/presto/connector/system/QuerySystemTable.java]: toBlock sounds pretty generic - can you make this function name more descriptive? e.g. `resourceGroupIdToBlock`

File: presto-main/src/main/resources/webapp/assets/query.js [hunk 4]
```diff
@@ -1118,7 +1118,7 @@ let QueryDetail = React.createClass({
                                         Resource Group
                                     </td>
                                     <td className="info-text wrap-text">
-                                        { query.resourceGroupName }
+                                        { formatResourceGroupId(query.resourceGroupId) }
```
Comment (raghavsethi) [inline presto-main/src/main/resources/webapp/assets/query.js]: Just put the join inline.

**Pull request merged**